### DISCUSSION TITLE
Help Connect Rpi zero W to nanoVNA-H

### DISCUSSION THREAD

**Discussion by ngriesert** [Ho-Ro/nanovna-tools#6] [Q&A]
Hello, 
    I am trying to interface an RPi zero W to a nanoVNA-H. I am reading through the files in nanovna-tools-main. I very well may not have set something up correctly, but I would really like to figure out how to make this work. Here is the error that I see:

ngriesert@antennapi:~/nanovna-tools-main $ python nanovna_communication_template.py -D
Traceback (most recent call last):
  File "/home/ngriesert/nanovna-tools-main/nanovna_communication_template.py", line 53, in <module>
    nanodevice = options.device or getdevice()
  File "/home/ngriesert/nanovna-tools-main/nanovna_communication_template.py", line 32, in getdevice
    raise OSError("device not found")
OSError: device not found

I have the single micro-usb port from the rip connected to the usb port of the nanoVNA-H.

Thank you for any feedback.
ng

**Comment by Ho-Ro** [comment]
Is the NanoVNA detected with `lsusb`?
What does `dmesg` report when you plug it in?

**Reply by ngriesert** [reply]
I don’t know what lsusb or dmesg are. I copied all of the files from this repository over to the Rpi zero and ran the script shown in the post. If there is some documentation or archived discussions that describe debugging these issues please advise. I am not experienced with this. I made an antenna that has wireless tuning via two stepper motors controlled by a python script and an Rpi zero w. I want to change the tuning from a manual process to an automatic process, but I need to be able to run sweeps, search the SWR, tune the antenna with motor inputs and repeat until optímal form desired freq. The motor control was easy with gpio control, but interfacing and extracting data from the NanoVNA H is proving more difficult.

**Reply by ngriesert** [reply]
I see. I will try lsusb and dmesg.

**Reply by Ho-Ro** [reply]
I'm currently not at my home desk during the week and have no NanoVNA at hand to provide a step-by-step setup with RasPi.
